theseus/optimizer/sparse_linearization.py

```diff
@@ -152,12 +158,33 @@ def _atb_impl(self) -> torch.Tensor:
             A_col_ind = A_row_ptr.new_tensor(self.A_col_ind)
 
             # unsqueeze at the end for consistency with DenseLinearization
-            self._Atb = tmat_vec(
-                self.objective.batch_size,
+            self._Atb = sparse_mtv(
                 self.num_cols,
                 A_row_ptr,
                 A_col_ind,
-                self.A_val,
-                self.b,
+                self.A_val.double(),
+                self.b.double(),
             ).unsqueeze(2)
-        return self._Atb
+        return self._Atb.to(dtype=self.A_val.dtype)
+
+    def Av(self, v: torch.Tensor) -> torch.Tensor:
+        A_row_ptr = torch.tensor(self.A_row_ptr, dtype=torch.int32).to(
```

Comment:
Why `dtype=torch.int32`? How about `self.A_row_ptr.clone()` or `v.new_tensor(self.A_row_ptr)`?

Reply:
We can ignore this, this is already merged into main. It was showing up because I hadn't rebased yet.